plugin-core/src/main/java/appland/actions/QuickReviewAction.java

```diff
@@ -0,0 +1,267 @@
+            private List<GitRef> refs;
+            private boolean dirty = false;
+
+            @Override
+            public void run(@NotNull ProgressIndicator indicator) {
+                try {
+                    refs = getItems(project, repository);
+                    dirty = isDirty();
+                } catch (VcsException ex) {
+                    throw new RuntimeException("Failed to fetch Git refs", ex);
+                }
+            }
+
+            private boolean isDirty() {
+                var handler = new GitLineHandler(project, repository.getRoot(), GitCommand.STATUS);
+                handler.setSilent(true);
+                handler.addParameters("--porcelain");
+                try {
+                    var result = Git.getInstance().runCommand(handler);
+                    result.throwOnError();
+                    return !result.getOutput().isEmpty();
+                } catch (VcsException e) {
+                    return false; // If we can't check, assume not dirty
+                }
+            }
+
+            @Override
+            public void onSuccess() {
+                if (refs == null || refs.isEmpty()) {
+                    return;
+                }
+
+                var head = repository.getInfo().getCurrentRevision();
+                var lastPickedRef = PropertiesComponent.getInstance().getValue(LAST_PICKED_REF_KEY);
+                AtomicBoolean seenLastPicked = new AtomicBoolean(false);
+
+                refs = refs.stream()
+                        /* only show HEAD if the repository is dirty */
+                        .filter(gitRef -> dirty || !gitRef.commit.equals(head))
+                        .peek(gitRef -> {
+                            if (gitRef.commit.equals(head)) gitRef.description = "review uncommitted changes";
+                            if (gitRef.label.equals(lastPickedRef)) {
+                                gitRef.description = "last picked ⋅ " + gitRef.description;
+                                seenLastPicked.set(true);
+                            }
+                        })
+                        .sorted(Comparator.comparing((GitRef gitRef) ->
+                                // if we have seen the last picked ref, sort it to the start
+                                // otherwise show common main branches first
+                                seenLastPicked.get() ? !gitRef.label.equals(lastPickedRef)
+                                        : !COMMON_MAIN_BRANCHES.contains(gitRef.label)))
+                        .toList();
+
+                var popup = JBPopupFactory.getInstance()
+                        .createPopupChooserBuilder(refs)
+                        .setRenderer(new GitRefCellRenderer())
+                        .setTitle("Select base for review")
+                        .setMovable(false)
+                        .setResizable(false)
+                        .setRequestFocus(true)
```

Comment:
Perhaps give the user some more control and let the IDE remember the dimensions?

```suggestion
                        .setMovable(true)
                        .setResizable(true)
                        .setDimensionServiceKey("appland.action.quickReview")
```